### PR TITLE
Remove use of deprecated STSC 'Lock'

### PR DESCRIPTION
**Pull request by artemcm**
- SwiftScan use of it seems not needed anymore
- ArgsResolver use can use 'NSLock' directly

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Review by cltnschlosser — APPROVED** [review]

File: Sources/SwiftDriver/Execution/ArgsResolver.swift [hunk 1]
```diff
@@ -202,3 +203,12 @@ public final class ArgsResolver {
     }
   }
 }
+
+fileprivate extension NSLock {
+    /// Execute the given block while holding the lock.
+    func withLock<T> (_ body: () throws -> T) rethrows -> T {
```
Comment (compnerd) [inline Sources/SwiftDriver/Execution/ArgsResolver.swift:209]: Do we want to add a note to keep this in sync with SPM?

**Review by compnerd — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci test

**Pull request merged**